### PR TITLE
TreeSheets master version

### PR DESCRIPTION
**Pull request by hamgom95**
Flatpak for TreeSheets hierarchical note taking app.

Uses the latest TreeSheets version from its git master branch.

### CODE REVIEW

File: org.aardappel.TreeSheets.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        dimension; and like a text editor, but with structure.
+    </p>
+    <p xml:lang="de">
+        Das Programm teilt sich mehrere Charakteristiken mit bekannteren Werkzeugen und erweitert diese.
+        Wie eine Tabellenkalkulation, jedoch mit hierarchischen Feldern. Wie ein Texteditor, jedoch mit Struktur.
+    </p>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image>http://strlen.com/treesheets/docs/images/screenshots/screenshot_todo_half.png</image>
```
Comment (AsciiWolf) [inline org.aardappel.TreeSheets.appdata.xml]: Isn't there any Linux screenshot available?

Comment (hamgom95) [reply org.aardappel.TreeSheets.appdata.xml]: You are right, I completely missed that. I replaced the screenshot with a linux one.

File: org.aardappel.TreeSheets.json [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    "branch": "master",
+    "tags": ["master", "notes", "wxWidgets"],
+    "build-options": {
+        "cflags": "-O2 -g",
+        "cxxflags": "-O2 -g",
+        "env": {
+            "V": "1"
+        },
+        "build-args": [
+            "--share=network"
```
Comment (TingPing) [inline org.aardappel.TreeSheets.json]: This can't work, all builds are offline.

File: org.aardappel.TreeSheets.json [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        {
+            "name": "wxWidgets",
+            "config-opts": [
+                "--enable-unicode"
+            ],
+            "cxxflags": [
+                "-std=c++17"
+            ],
+            "make-args": [
+                "-j4"
```
Comment (TingPing) [inline org.aardappel.TreeSheets.json]: Don't set this.

File: org.aardappel.TreeSheets.json [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        "*.la",
+        "*.a"
+    ],
+    "modules": [
+        {
+            "name": "wxWidgets",
+            "config-opts": [
+                "--enable-unicode"
+            ],
+            "cxxflags": [
```
Comment (TingPing) [inline org.aardappel.TreeSheets.json]: This should print a warning about an invalid property. This goes in `build-options`, see `man flatpak-manifest`

File: org.aardappel.TreeSheets.json [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                {
+                    "type": "archive",
+                    "url": "https://github.com/wxWidgets/wxWidgets/releases/download/v3.1.0/wxWidgets-3.1.0.tar.bz2",
+                    "sha256": "e082460fb6bf14b7dd6e8ac142598d1d3d0b08a7b5ba402fdbf8711da7e66da8"
+                }
+            ]
+        },
+        {
+            "name": "treesheets",
+            "buildsystem": "cmake",
```
Comment (TingPing) [inline org.aardappel.TreeSheets.json]: `cmake-ninja`

File: org.aardappel.TreeSheets.json [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+            "make-args": [
+                "-j8"
```
Comment (TingPing) [inline org.aardappel.TreeSheets.json]: Don't set this.

File: org.aardappel.TreeSheets.json [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+                    "url": "https://github.com/aardappel/treesheets"
+                }
+            ]
+        },
+        {
+            "name": "metadata",
+            "buildsystem": "simple",
+            "build_commands": [
+                "install -Dm644 org.aardappel.TreeSheets.appdata.xml /app/share/metainfo/org.aardappel.TreeSheets.appdata.xml",
+                "install -Dm644 org.aardappel.TreeSheets.desktop /app/share/applications/org.aardappel.TreeSheets.desktop"
```
Comment (TingPing) [inline org.aardappel.TreeSheets.json]: Upstream has a desktop file, use it.

File: org.aardappel.TreeSheets.json [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    "finish-args": [
+        "--socket=x11",
+        "--share=ipc",
+        "--filesystem=xdg-documents/treesheets:create",
+
+        /* Shell detection and extension management */
+        "--talk-name=org.gnome.SessionManager",
+
+        /* dconf */
+        "--filesystem=/usr/share/glib-2.0/schemas:ro",
```
Comment (TingPing) [inline org.aardappel.TreeSheets.json]: Remove this.

File: org.aardappel.TreeSheets.json [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+            ],
+            "post-install": [
+                "install -Dm644 TS/images/icon16.png /app/share/icons/hicolor/16x16/apps/org.aardappel.TreeSheets.png",
+                "install -Dm644 TS/images/icon32.png /app/share/icons/hicolor/32x32/apps/org.aardappel.TreeSheets.png",
+                "install -m755 TS/treesheets /app/"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/aardappel/treesheets"
```
Comment (TingPing) [inline org.aardappel.TreeSheets.json]: Flathub only accepts versioned and stable releases. If this project has neither then this package can wait or you can worth with upstream to get a release maybe.

File: org.aardappel.TreeSheets.appdata.xml [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+        Das Programm teilt sich mehrere Charakteristiken mit bekannteren Werkzeugen und erweitert diese.
+        Wie eine Tabellenkalkulation, jedoch mit hierarchischen Feldern. Wie ein Texteditor, jedoch mit Struktur.
+    </p>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image>http://strlen.com/treesheets/docs/images/screenshots/screenshot_todo_linux.png</image>
+    </screenshot>
+  </screenshots>
+</component>
```
Comment (TingPing) [inline org.aardappel.TreeSheets.appdata.xml]: Missing `releases`.

File: org.aardappel.TreeSheets.appdata.xml [hunk 11]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline org.aardappel.TreeSheets.appdata.xml]: Have you submitted it upstream?

File: org.aardappel.TreeSheets.json [hunk 12]
```diff
@@ -0,0 +1,88 @@
+{
+    "app-id": "org.aardappel.TreeSheets",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "18.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "/app/treesheets",
+    "branch": "master",
+    "tags": ["master", "notes", "wxWidgets"],
```
Comment (barthalion) [inline org.aardappel.TreeSheets.json]: What's the point of these tags?

File: org.aardappel.TreeSheets.json [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+                    "path": "org.aardappel.TreeSheets.desktop"
+                },
+                {
+                    "type": "file",
+                    "path": "org.aardappel.TreeSheets.appdata.xml"
+                }
+            ]
+        }
+    ],
+    "finish-args": [
```
Comment (barthalion) [inline org.aardappel.TreeSheets.json]: Please move these above sources array.

**Comment by barthalion** [comment]
Hi @aardappel, would you be interested in helping out with maintenance or at least having RW access?

File: org.aardappel.TreeSheets.json [hunk 14]
```diff
@@ -0,0 +1,88 @@
+{
+    "app-id": "org.aardappel.TreeSheets",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "18.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "/app/treesheets",
+    "branch": "master",
```
Comment (TingPing) [inline org.aardappel.TreeSheets.json]: Must be `"stable"`.

**Comment by nedrichards** [comment]
bot, build org.aardappel.TreeSheets

**Comment by barthalion** [comment]
I'm closing it due to inactivity – but please re-open it or resubmit as new pull request if you would like to see it merged in the future.

**Pull request closed without merging**